src/libstore/store-api.cc

```diff
@@ -589,14 +589,29 @@ void copyStorePath(ref<Store> srcStore, ref<Store> dstStore,
     uint64_t total = 0;
 
     // FIXME
-#if 0
     if (!info->narHash) {
+        auto progress = [&](size_t len) {
+            total += len;
+            act.progress(total, info->narSize);
+        };
+        struct MyStringSink : StringSink
+        {
+            typedef std::function<void(size_t)> Callback;
+            Callback callback;
+            MyStringSink(Callback callback) : callback(callback) { }
+            void operator () (const unsigned char * data, size_t len) override
+            {
+                StringSink::operator ()(data, len);
+                callback(len);
+            };
+        };
+        MyStringSink sink(progress);
+        srcStore->narFromPath({storePath}, sink);
```

Comment:
Note that this will cause the NAR to be copied twice (due to the later call to `narFromPath`). You could create a `StringSource` for `sink.s` and pass it to `dstStore->addToStore()`.

BTW I think the whole progress stuff can be removed here since this is just a backward compatibility path.